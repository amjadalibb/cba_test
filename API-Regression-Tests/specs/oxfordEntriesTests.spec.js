const { requestAPI } = require('../support/api.js');
const { expect } = require('chai');
const { expectSnapshot, httpStatus, report } = require('../support');
const { SCENARIOS } = require('../tests/oxfordEntriesTests.js')

describe('Oxford Dictionary Tests for Entries', function() {
	it('Get entries of a word Insurance', async function() {
		report(this, 'API Options BaseURL', SCENARIOS.optionsInvalid.baseURL);
		report(this, 'API Options Endpoint', SCENARIOS.optionsValid.endpoint);
		const response = await requestAPI(SCENARIOS.optionsValid);
        report(this, 'Origin', response.data.results[0].lexicalEntries[0].entries[0].etymologies)
		expectSnapshot(response.data.results[0].lexicalEntries[0].entries[0].etymologies).toMatchSnapshot(this);
	});

	it('Get entries of a invalid word ThisWordDoesNotExist', async function() {
		report(this, 'API Options BaseURL', SCENARIOS.optionsInvalid.baseURL);
		report(this, 'API Options Endpoint', SCENARIOS.optionsInvalid.endpoint);
		const response = await requestAPI(SCENARIOS.optionsInvalid);
		report(this, 'API response', response.data);
		expect(response.status).to.equal(httpStatus.NOT_FOUND);
		expectSnapshot(response.data).toMatchSnapshot(this);
	});

	it('Bad request on getting entries of word', async function() {
		report(this, 'API Options BaseURL', SCENARIOS.optionsBadRequest.baseURL);
		report(this, 'API Options Endpoint', SCENARIOS.optionsBadRequest.endpoint);
		report(this, 'API Options Params', SCENARIOS.optionsBadRequest.params);
		const response = await requestAPI(SCENARIOS.optionsBadRequest);
		report(this, 'API response', response.data);
		expect(response.status).to.equal(httpStatus.BAD_REQUEST);
		expectSnapshot(response.data).toMatchSnapshot(this);
	});
});