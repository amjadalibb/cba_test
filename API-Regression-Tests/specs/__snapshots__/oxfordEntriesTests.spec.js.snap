// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oxford Dictionary Tests for Entries Bad request on getting entries of word 1`] = `
Object {
  "error": "Unknown filter value 'UnknownFilter' for language 'en'. Please, consult valid values in the utility endpoints.",
}
`;

exports[`Oxford Dictionary Tests for Entries Get entries of a invalid word ThisWordDoesNotExist 1`] = `
Object {
  "error": "No entry found matching supplied source_lang, word and provided filters",
}
`;

exports[`Oxford Dictionary Tests for Entries Get entries of a word Insurance 1`] = `
Array [
  "late Middle English (originally as ensurance in the sense ‘ensuring, assurance, a guarantee’): from Old French enseurance, from enseurer (see ensure). insurance (sense 1) dates from the mid 17th century",
]
`;
